app/hooks/handle_send_message.ts

```diff
@@ -0,0 +1,223 @@
+import {Events, Screens} from '@constants';
+import {NOTIFY_ALL_MEMBERS} from '@constants/post_draft';
+import {useServerUrl} from '@context/server';
+import DraftUploadManager from '@managers/draft_upload_manager';
+import * as DraftUtils from '@utils/draft';
+import {isReactionMatch} from '@utils/emoji/helpers';
+import {getFullErrorMessage} from '@utils/errors';
+import {preventDoubleTap} from '@utils/tap';
+import {confirmOutOfOfficeDisabled} from '@utils/user';
+
+import type CustomEmojiModel from '@typings/database/models/servers/custom_emoji';
+
+type Props = {
+    value: string;
+    channelId: string;
+    rootId: string;
+    maxMessageLength: number;
+    files: FileInfo[];
+    customEmojis: CustomEmojiModel[];
+    enableConfirmNotificationsToChannel?: boolean;
+    useChannelMentions: boolean;
+    membersCount: number;
+    userIsOutOfOffice: boolean;
+    currentUserId: string;
+    channelType: ChannelType | undefined;
+    postPriority: PostPriority;
+    clearDraft?: () => void;
+}
+
+export const useHandleSendMessage = ({
+    value,
+    channelId,
+    rootId,
+    files,
+    maxMessageLength,
+    customEmojis,
+    enableConfirmNotificationsToChannel,
+    useChannelMentions,
+    membersCount = 0,
+    userIsOutOfOffice,
+    currentUserId,
+    channelType,
+    postPriority,
+    clearDraft,
+}: Props) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const [sendingMessage, setSendingMessage] = useState(false);
+    const [channelTimezoneCount, setChannelTimezoneCount] = useState(0);
+
+    const canSend = useCallback(() => {
+        if (sendingMessage) {
+            return false;
+        }
+
+        const messageLength = value.trim().length;
+
+        if (messageLength > maxMessageLength) {
+            return false;
+        }
+
+        if (files.length) {
+            const loadingComplete = !files.some((file) => DraftUploadManager.isUploading(file.clientId!));
+            return loadingComplete;
+        }
+
+        return messageLength > 0;
+    }, [sendingMessage, value, files, maxMessageLength]);
+
+    const handleReaction = useCallback((emoji: string, add: boolean) => {
+        handleReactionToLatestPost(serverUrl, emoji, add, rootId);
+        clearDraft?.();
+        setSendingMessage(false);
+    }, [serverUrl, rootId, clearDraft]);
+
+    const doSubmitMessage = useCallback(() => {
+        const postFiles = files.filter((f) => !f.failed);
+        const post = {
+            user_id: currentUserId,
+            channel_id: channelId,
+            root_id: rootId,
+            message: value,
+        } as Post;
+
+        if (!rootId && (
+            postPriority.priority ||
+            postPriority.requested_ack ||
+            postPriority.persistent_notifications)
+        ) {
+            post.metadata = {
+                priority: postPriority,
+            };
+        }
+
+        createPost(serverUrl, post, postFiles);
```

Comment:
Same here.  Should we worry if the post is created or not?

Reply:
This can also be taken up later.